conftest.py

```diff
@@ -88,3 +88,13 @@ def pytest_runtest_setup(item):
 def pytest_runtest_teardown(item, nextitem):
     if isinstance(item, DoctestItem):
         set_config(print_changed_only=False)
+
+
+# We don't want pytest to run these files since they immediately raise a
+# DeprecationWarning and that would make CI unhappy.
+# These files don't contain any real code, they just raise a warning. We still
+# ensure these warnings are properly raised in the tests.
+collect_ignore_glob = [
+    "sklearn/neural_network/rbm.py",  # 0.24
+    "sklearn/neural_network/multilayer_perceptron.py",  # 0.24
```

Comment:
I would still prefer the approach as outlined in https://github.com/scikit-learn/scikit-learn/pull/14913#discussion_r322889700